Enhance network hostname documentation. (#13524)
zephyrproject-rtos/zephyr: doc/reference/networking/net_hostname.rst

Overview
********

> A networked device might need a hostname in certain cases like if the device

Reviewer: ```suggestion
A networked device might need a hostname, for example, if the device
```

is configured to be a mDNS responder (see :ref:`dns_resolve_interface` for
details) and needs to respond to ``<hostname>.local`` DNS queries.

> The :option:`CONFIG_NET_HOSTNAME_ENABLE` must be set in config file in order

Reviewer: set in which config file?  or should we just delete "in config file"

Author: I will delete the "in config file" part, it is useless.

to store the hostname and enable the relevant APIs. If the option is enabled,
then the default hostname is set to be ``zephyr`` by
:option:`CONFIG_NET_HOSTNAME` option.

If the same firmware image is used to flash multiple boards, then it is not
practical to use the same hostname in all of the boards. In that case, one
can enable :option:`CONFIG_NET_HOSTNAME_UNIQUE` which will add a unique
postfix to the hostname. By default link address of the first network interface
is used as a postfix. In Ethernet networks, the link address refers to MAC
address. For example, if the link address is ``01:02:03:04:05:06``, then the
unique hostname could be ``zephyr010203040506``. If you want to set the prefix

> yourself, then make sure to call ``net_hostname_set_postfix()`` before the

Reviewer: ```suggestion
yourself, then call ``net_hostname_set_postfix()`` before the
```
